### PR TITLE
Protect against invalid opposing edge in the opposing method. 

### PR DESCRIPTION
**Pull request by dnesbitt61**
This fixes a crash occurring in production - leading to the edge export file being much smaller than it should be.

### CODE REVIEW

**Comment by dnesbitt61** [comment]
Fixes #578

File: src/valhalla_export_edges.cc [hunk 1]
```diff
@@ -223,15 +232,18 @@ int main(int argc, char *argv[]) {
...
-      //get the opposing edge as well
-      edge_t opposing_edge = opposing(reader, tile, edge);
-      edge_set.set(tile_set.find(opposing_edge.i.Tile_Base())->second + opposing_edge.i.id());
-      ++set;
-
       //shortcuts arent real and maybe we dont want ferries
       if(edge.e->is_shortcut() || (!ferries && edge.e->use() == Use::kFerry))
         continue;
 
+      //get the opposing edge as well (ensure a valid edge is returned)
```
Comment (kevinkreiser) [inline src/valhalla_export_edges.cc]: why is this moved below the `continue`? doing so makes it so that the `edge_set.set` and `++set` dont happen, and so for shortcuts and ferries we will have to look at and mark them in additional loop interations

File: src/valhalla_export_edges.cc [hunk 2]
```diff
@@ -51,13 +51,22 @@ struct edge_t {
...
-    tile = reader.GetGraphTile(edge->endnode());
+  const GraphTile* t = edge->leaves_tile() ? reader.GetGraphTile(edge->endnode()) : tile;
   auto id = edge->endnode();
-  auto* node = tile->node(edge->endnode());
-  id.fields.id = tile->node(id)->edge_index() + edge->opp_index();
-  return {id, tile->directededge(id)};
+  id.fields.id = t->node(id)->edge_index() + edge->opp_index();
+
+  // Check for invalid opposing index
+  if (edge->opp_index() == kMaxEdgesPerNode) {
```
Comment (kevinkreiser) [inline src/valhalla_export_edges.cc]: so this is valid data right? why does this happen? a node with too many outgoing edges? we cant get opposing edges for the 9th edge leaving a node or something?

File: src/valhalla_export_edges.cc (hunk 1 above)
Comment (dnesbitt61) [inline src/valhalla_export_edges.cc]: Could go either way on this - generally the cost of getting an opposing edge can be high if it reaches into a neighboring tile. I can move this back if you think it makes sense. Probably doesn't make a lot of performance diff though we could check that later.

File: src/valhalla_export_edges.cc (hunk 2 above)
Comment (dnesbitt61) [inline src/valhalla_export_edges.cc]: investigating why this happens. It should never happen. We allow up to 127 edges. If an error occurs the opp_index can end up being set to 127 (and an error is logged).

File: src/valhalla_export_edges.cc (hunk 1 above)
Comment (kevinkreiser) [inline src/valhalla_export_edges.cc]: ah ok. well if thats the case then it may as well be moved all the way down passed the last continue where it checks for names or not. i just want to make sure that nothing gets double counted haha! and i think that would be the case regardless of where this goes right?

**Comment by kevinkreiser** [comment]
:rocket:
